File: exercises/java/src/main/java/cards/AnimalCard.java
package cards;

public class AnimalCard {

    private final Animal animal;

    public AnimalCard(Animal animal) {
        this.animal = animal;
    }

    public boolean snap( AnimalCard otherCard ){
        return otherCard != null && this.animal.equals(otherCard.animal);
    }

    @Override
    public String toString() {
        return animal.toString();
    }
}

File: exercises/typescript/src/cards/animal-card.ts
export class AnimalCard {
  private readonly animal: string

  constructor(animal: string) {
    this.animal = animal
  }

  snap(otherCard: AnimalCard): boolean {
    return otherCard && this.animal === otherCard.animal
  }

  toString(): string {
    return this.animal.toString()
  }
}